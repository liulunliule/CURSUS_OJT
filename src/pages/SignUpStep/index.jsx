import React, { useState } from "react";
import "./index.scss";
import { logo, logo_Small } from "../../assets";
import { Link, useNavigate } from "react-router-dom";
import { message, Select } from "antd";
import TextArea from "antd/es/input/TextArea";

function SignUpStep() {
    const [isOpen, setIsOpen] = useState(true);
    

    const handleCloseSelect = () => {
        setIsOpen(false);
    };

    const handleOpenSelect = () => {
        setIsOpen(true);
    };

    return (
        <div>
            <div className="signUpStep">
                <img src={logo} alt="" className="signUpStep__logo" />
                <div className="signUpStep__main main">
                    <div className="signUpStep__switch-wrap">
                        <div className="signUpStep__switch">
                            <input
                                type="radio"
                                name="signUpOption"
                                id="signUpStep-switch"
                                className="signUpStep__switch-input"
                                hidden
                                defaultChecked
                                onClick={handleOpenSelect}
                            />
                            <label
                                htmlFor="signUpStep-switch"
                                className="signUpStep__switch-label"
                            >
                                <span className="signUpStep__switch-value">
                                    Instructor Sign Up
                                </span>
                            </label>
                        </div>
                        <div className="signUpStep__switch">
                            <input
                                type="radio"
                                name="signUpOption"
                                id="signUpStep-switch2"
                                className="signUpStep__switch-input"
                                hidden
                                onClick={handleCloseSelect}
                            />
                            <label
                                htmlFor="signUpStep-switch2"
                                className="signUpStep__switch-label"
                            >
                                <span className="signUpStep__switch-value">
                                    Student Sign Up
                                </span>
                            </label>
                        </div>
                    </div>
                    {/* <p className="signUpStep__desc">Sign Up and Create Course!</p> */}
                    <form className="signUpStep__form">
                        {isOpen && (
                            <div className="signUpStep__select">
                                <Select
                                    className="custom-select"
                                    showSearch
                                    style={{
                                        width: "100%",
                                    }}
                                    placeholder="Search Category"
                                    options={[
                                        {
                                            value: "1",
                                            label: "Development",
                                        },
                                        {
                                            value: "2",
                                            label: "Business",
                                        },
                                        {
                                            value: "3",
                                            label: "Finance & Accounting",
                                        },
                                        {
                                            value: "4",
                                            label: "IT & Software",
                                        },
                                        {
                                            value: "5",
                                            label: "Office Productivity",
                                        },
                                        {
                                            value: "6",
                                            label: "Design",
                                        },
                                        {
                                            value: "7",
                                            label: "Marketing",
                                        },
                                        {
                                            value: "8",
                                            label: "Lifestyle",
                                        },
                                        {
                                            value: "9",
                                            label: "Photography",
                                        },
                                        {
                                            value: "10",
                                            label: "Health & Fitness",
                                        },
                                        {
                                            value: "11",
                                            label: "Music",
                                        },
                                        {
                                            value: "12",
                                            label: "Teaching and Academic",
                                        },
                                    ]}
                                />
                            </div>
                        )}
                        <div className="signUpStep__textArea">
                            <TextArea
                                rows={3}
                                placeholder="Write a little description about you..."
                                maxLength={12000}
                            />
                        </div>
                        <p className="signUpStep__limit">
                            Your biography should have at least 12000
                            characters.
                        </p>
                        <button className="signUpStep__btn button">
                            Instructor Sign Up Now
                        </button>
                        <div className="signUpStep__signup">
                            Already have an account?
                            <Link to="/login">Log In</Link>
                        </div>
                    </form>
                </div>

                <div className="signup__footer">
                    <img src={logo_Small} alt="" />
                    <p className="login__footer-content">
                        Â© 2024 <b>Cursus</b>. All Rights Reserved.
                    </p>
                </div>
            </div>
        </div>
    );
}

export default SignUpStep;
