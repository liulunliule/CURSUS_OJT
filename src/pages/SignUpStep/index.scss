.signUpStep {
    display: flex;
    position: relative;
    flex-direction: column;
    align-items: center;
    padding: 50px 0;
    background-color: #f7f7f7;
    z-index: 1;
    &::before {
        content: "";
        background: url(../../assets/img/backgroundLDP.svg) no-repeat center;
        inset: 0;
        position: absolute;
        background-size: cover;
        opacity: 0.04;
        z-index: -1;
    }
    &__logo {
        width: 135px;
    }

    // &__switch {
    //     &-label {
    //         position: relative;
    //         display: flex;
    //         justify-content: space-between;
    //         align-items: center;
    //         width: 478px;
    //         height: 56px;
    //         border: 1px solid #e5e5e5;
    //         border-radius: 5.25px;
    //         font-size: 14px;
    //         font-weight: 500;
    //         line-height: 24px;
    //         padding: 0 10px;
    //         background-color: #f7f7f7;
    //         user-select: none;

    //         &::before {
    //             position: absolute;
    //             content: "";
    //             bottom: 0;
    //             left: 0;
    //             width: 239px;
    //             height: 2px;
    //             background-color: #ed2a26;
    //         }
    //         &::after {
    //             position: absolute;
    //             content: "";
    //             top: 0;
    //             left: 239px;
    //             width: 0.1px;
    //             height: 53px;
    //             background-color: #e5e5e5;
    //         }
    //     }
    //     &-value {
    //         color: #ed2a26;
    //     }
    //     &-input:checked + &-label::before {
    //         width: 238px;
    //         left: auto;
    //         right: 0;
    //     }
    //     &-input:not(:checked)
    //         + &-label
    //         .signUpStep__switch-value:nth-of-type(2),
    //     &-input:checked + &-label .signUpStep__switch-value:nth-of-type(1) {
    //         color: #333;
    //     }
    // }
    &__switch:nth-of-type(1) .signUpStep__switch-label {
        padding: 0 65px;
    }
    &__switch:nth-of-type(2) .signUpStep__switch-label {
        padding: 0 70px;
    }
    &__switch {
        &-wrap {
            display: flex;
        }

        &-label {
            position: relative;
            display: flex;
            justify-content: space-between;
            align-items: center;
            width: 239px;
            height: 56px;
            border: 1px solid #e5e5e5;
            font-size: 14px;
            font-weight: 500;
            line-height: 24px;
            background-color: #f7f7f7;
            user-select: none;
            white-space: nowrap;

            &::before {
                position: absolute;
                content: "";
                bottom: 0;
                left: 0;
                width: 239px;
                height: 2px;
            }
        }
        &-value {
            color: #333;
        }
        &-input:checked + &-label::before {
            width: 238px;
            background-color: #ed2a26;
        }
        &-input:not(:checked)
            + &-label
            .signUpStep__switch-value:nth-of-type(2),
        &-input:checked + &-label .signUpStep__switch-value:nth-of-type(1) {
            color: #ed2a26;
        }
    }
    &__desc {
        margin-top: 49px;
        font-size: 14px;
        font-weight: 400;
        font-family: "Roboto", sans-serif;
        color: #686f7a;
    }

    &__form {
        margin-top: 47px;
    }

    &__textArea {
        margin-top: 25px;
    }

    &__limit {
        text-align: start;
        font-size: 11px;
        margin-top: 5px;
        color: #686f7a;
        font-size: 12px;
        font-family: "Roboto", sans-serif;
    }

    &__btn {
        margin-top: 30px;
        width: 479px;
        font-size: 14px;
        font-weight: 500;
    }

    &__signup {
        font-size: 14px;
        margin-top: 47px;
        font-weight: 400;
        font-family: "Roboto", sans-serif;
        color: #686f7a;
        text-align: center;
        line-height: 26px;

        a {
            margin-left: 4px;
            color: #4183c4;
            text-decoration: none;
        }

        & a:hover {
            color: #686f7a;
        }
    }
    &__footer {
        display: flex;
        align-items: center;
        margin-top: 57px;

        &-content {
            margin: 0;
        }
    }
}

//Antd custom css

.custom-select .ant-select-selection-placeholder {
    display: flex;
    justify-content: flex-start;

    color: #00b2a9;
    font-size: 14px;
    font-weight: 400;
    font-family: "Roboto", sans-serif;
}

.custom-select .ant-select-selector {
    display: flex;
    align-items: center;
    padding: 20px 15px !important;
}

.custom-select .ant-select-arrow {
    top: 20px;
}
