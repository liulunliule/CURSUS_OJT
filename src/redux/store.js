import { combineReducers, configureStore } from "@reduxjs/toolkit";
import savedCourseReducer from "./features/savedCourseSlice";
import blogReducer from './features/ourBlogSlice';
import officesReducer from './features/officesSlice';
import pressReducer from './features/pressSlice';
import payoutReducer from './features/payoutSlice';
import creditsReducer from "./features/creditsSlice";
import certificateReducer from "./features/myCertificateSlice";
import myProfileReducer from "./features/myProfileSlice";
import myHeaderReducer from "./features/myHeaderSlice";
import mySearchReducer from "./features/mySearchSlice";
import myStudioReducer from "./features/myStudioSlice";
import allInstructorReducer from "./features/allInstructorSlice";
import reviewFeedbackReducer from "./features/reviewfeedbackSlice";
import feedbackReportReducer from "./features/feedbackvsReportSlice";
import notificationsSettingReducer from "./features/NotificationsSettingSlice";
import privacySettingReducer from "./features/PrivacySettingSlice";
import userSlice from "./features/userSlice";
import certificationTestPageReducer from "./features/certificationTestPageSlice";
import certificationTestResultReducer from "./features/certificationTestResultSlice";
import analyticsReducer from "./features/analyticsSlice";
import storage from "redux-persist/lib/storage";
import { persistReducer, persistStore } from "redux-persist";
import courseSlice from "./features/courseSlice";
import checkoutReducer from './features/checkoutSlice';
import earningReducer from './features/earningSlice';
import statementReducer from './features/statementSlice';
import shoppingCartReducer from './features/shoppingCartSlice';

const persistConfig = {
    key: "root",
    storage,
    whitelist: [
        "user",
        "savedCourse",
        "certificate",
        "course",
        "myProfile",
        "myHeader",
        "mySearch",
        "myStudio",
        "allInstructor",
        "reviewFeedback",
        "feedbackReport",
        "notificationsSetting",
        "privacySetting",
    ],
};

const rootReducer = combineReducers({
    user: userSlice,
    savedCourse: savedCourseReducer,
    certificate: certificateReducer,
    certificationTestPage: certificationTestPageReducer,
    certificationTestResult: certificationTestResultReducer,
    analytics: analyticsReducer,
    course: courseSlice,
    myProfile: myProfileReducer,
    myHeader: myHeaderReducer,
    mySearch: mySearchReducer,
    myStudio: myStudioReducer,
    allInstructor: allInstructorReducer,
    reviewFeedback: reviewFeedbackReducer,
    feedbackReport: feedbackReportReducer,
    checkout: checkoutReducer,
    earnings: earningReducer,
    statement: statementReducer,
    credits: creditsReducer,
    payout: payoutReducer,
    blogs: blogReducer,
    offices: officesReducer,
    press: pressReducer,
    notificationsSetting: notificationsSettingReducer,
    privacySetting: privacySettingReducer,
    shoppingCart: shoppingCartReducer,
});

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
    reducer: persistedReducer,
});

export const persistor = persistStore(store);
