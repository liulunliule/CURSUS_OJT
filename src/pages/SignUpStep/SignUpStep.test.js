import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom'; // Import MemoryRouter
import { store } from '../../redux/store';
import SignUpStep from '.';

test('test SignUpStep Page', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <SignUpStep/>
      </MemoryRouter>
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
