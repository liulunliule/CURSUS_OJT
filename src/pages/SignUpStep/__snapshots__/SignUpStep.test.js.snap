// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test SignUpStep Page 1`] = `
<div>
  <div
    className="signUpStep"
  >
    <img
      alt=""
      className="signUpStep__logo"
      src="logo.svg"
    />
    <div
      className="signUpStep__main main"
    >
      <div
        className="signUpStep__switch-wrap"
      >
        <div
          className="signUpStep__switch"
        >
          <input
            className="signUpStep__switch-input"
            defaultChecked={true}
            hidden={true}
            id="signUpStep-switch"
            name="signUpOption"
            onClick={[Function]}
            type="radio"
          />
          <label
            className="signUpStep__switch-label"
            htmlFor="signUpStep-switch"
          >
            <span
              className="signUpStep__switch-value"
            >
              Instructor Sign Up
            </span>
          </label>
        </div>
        <div
          className="signUpStep__switch"
        >
          <input
            className="signUpStep__switch-input"
            defaultChecked={false}
            hidden={true}
            id="signUpStep-switch2"
            name="signUpOption"
            onClick={[Function]}
            type="radio"
          />
          <label
            className="signUpStep__switch-label"
            htmlFor="signUpStep-switch2"
          >
            <span
              className="signUpStep__switch-value"
            >
              Student Sign Up
            </span>
          </label>
        </div>
      </div>
      <form
        className="signUpStep__form"
        onSubmit={[Function]}
      >
        <div
          className="signUpStep__select"
        >
          <div
            className="ant-select ant-select-outlined custom-select css-dev-only-do-not-override-1pg9a38 ant-select-single ant-select-show-arrow ant-select-show-search"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            style={
              Object {
                "width": "100%",
              }
            }
          >
            <div
              className="ant-select-selector"
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="ant-select-selection-search"
              >
                <input
                  aria-autocomplete="list"
                  aria-controls="undefined_list"
                  aria-expanded={false}
                  aria-haspopup="listbox"
                  aria-owns="undefined_list"
                  autoComplete="off"
                  className="ant-select-selection-search-input"
                  disabled={false}
                  onChange={[Function]}
                  onCompositionEnd={[Function]}
                  onCompositionStart={[Function]}
                  onKeyDown={[Function]}
                  onMouseDown={[Function]}
                  onPaste={[Function]}
                  readOnly={false}
                  role="combobox"
                  style={
                    Object {
                      "opacity": null,
                    }
                  }
                  type="search"
                  unselectable={null}
                  value=""
                />
              </span>
              <span
                className="ant-select-selection-placeholder"
              >
                Search Category
              </span>
            </div>
            <span
              aria-hidden={true}
              className="ant-select-arrow"
              onMouseDown={[Function]}
              style={
                Object {
                  "WebkitUserSelect": "none",
                  "userSelect": "none",
                }
              }
              unselectable="on"
            >
              <span
                aria-label="down"
                className="anticon anticon-down ant-select-suffix"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="down"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                </svg>
              </span>
            </span>
          </div>
        </div>
        <div
          className="signUpStep__textArea"
        >
          <textarea
            className="ant-input css-dev-only-do-not-override-1pg9a38 ant-input-outlined"
            disabled={false}
            maxLength={12000}
            onBlur={[Function]}
            onChange={[Function]}
            onCompositionEnd={[Function]}
            onCompositionStart={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Write a little description about you..."
            rows={3}
            style={
              Object {
                "resize": undefined,
              }
            }
            value=""
          />
        </div>
        <p
          className="signUpStep__limit"
        >
          Your biography should have at least 12000 characters.
        </p>
        <button
          className="signUpStep__btn button"
          type="submit"
        >
          Sign Up Now
        </button>
        <div
          className="signUpStep__signup"
        >
          Already have an account?
          <a
            href="/login"
            onClick={[Function]}
          >
            Log In
          </a>
        </div>
      </form>
    </div>
    <div
      className="signup__footer"
    >
      <img
        alt=""
        src="small_logo.png"
      />
      <p
        className="login__footer-content"
      >
        Â© 2024 
        <b>
          Cursus
        </b>
        . All Rights Reserved.
      </p>
    </div>
  </div>
</div>
`;
